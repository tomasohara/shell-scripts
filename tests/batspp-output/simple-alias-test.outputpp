#!/usr/bin/env bats


# Make executables ./tests/../ visible to PATH
PATH="/home/aveey/tom-project/shell-scripts/tests/batspp-only/../:$PATH"

# Enable aliases
shopt -s expand_aliases



@test "test-1" {
	testfolder="/tmp/batspp-70644/test-1"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $"alias count-words='wc -w'\n" "=========="
	test-1-actual 
	echo "=========" $'' "========="
	test-1-expected 
	echo "============================"
	# ???: "alias count-words='wc -w'\n"=$(test-1-actual)
	# ???: ''=$(test-1-expected)
	[ "$(test-1-actual)" == "$(test-1-expected)" ]
}

function test-1-actual () {
	# no-op in case content just a comment
	true

	alias count-words='wc -w'

}

function test-1-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED

END_EXPECTED
}


@test "test-3" {
	testfolder="/tmp/batspp-70644/test-3"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'echo abc def ght | count-words\n' "=========="
	test-3-actual 
	echo "=========" $'3' "========="
	test-3-expected 
	echo "============================"
	# ???: 'echo abc def ght | count-words\n'=$(test-3-actual)
	# ???: '3'=$(test-3-expected)
	[ "$(test-3-actual)" == "$(test-3-expected)" ]
}

function test-3-actual () {
	# no-op in case content just a comment
	true

	echo abc def ght | count-words

}

function test-3-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
3
END_EXPECTED
}
