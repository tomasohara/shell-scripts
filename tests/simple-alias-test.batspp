# Simple alias test

# Global Setup
$ alias count-words='wc -w'

$ echo abc def ght | count-words
3
