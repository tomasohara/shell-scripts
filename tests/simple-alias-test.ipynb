{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Simple alias test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "alias count-words='wc -w'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n"
     ]
    }
   ],
   "source": [
    "echo abc def ght | count-words"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
