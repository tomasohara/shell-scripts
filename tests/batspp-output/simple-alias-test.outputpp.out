1..2
ok 1 test-1
not ok 2 test-3
# (from function `test-3-actual' in file batspp-output/simple-alias-test.outputpp, line 66,
#  in test file batspp-output/simple-alias-test.outputpp, line 53)
#   `test-3-actual ' failed with status 127
# ========== echo abc def ght | count-words
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/simple-alias-test.outputpp: line 66: count-words: command not found
